From 53a300cf3062b2721d83da424e995d2836b4ab7d Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:02:46 -0400
Subject: [fs] btrfs: filter invalid arg for btrfs resize

Message-id: <1407179109-13714-280-git-send-email-sandeen@redhat.com>
Patchwork-id: 87243
O-Subject: [RHEL7.1 PATCH 279/422] btrfs: filter invalid arg for btrfs resize
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Gui Hecheng <guihc.fnst@cn.fujitsu.com>

commit 9a40f1222a372de77344d85d31f8fe0e1c0e60e7
Author: Gui Hecheng <guihc.fnst@cn.fujitsu.com>
Date:   Mon Mar 31 18:03:25 2014 +0800

    btrfs: filter invalid arg for btrfs resize

    Originally following cmds will work:
     # btrfs fi resize -10A  <mnt>
     # btrfs fi resize -10Gaha <mnt>
    Filter the arg by checking the return pointer of memparse.

    Signed-off-by: Gui Hecheng <guihc.fnst@cn.fujitsu.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/ioctl.c b/fs/btrfs/ioctl.c
index 27d551c..86393b7 100644
--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -1472,6 +1472,7 @@ static noinline int btrfs_ioctl_resize(struct file *file,
  struct btrfs_trans_handle *trans;
  struct btrfs_device *device = NULL;
  char *sizestr;
+ char *retptr;
  char *devstr = NULL;
  int ret = 0;
  int mod = 0;
@@ -1539,8 +1540,8 @@ static noinline int btrfs_ioctl_resize(struct file *file,
    mod = 1;
    sizestr++;
   }
-  new_size = memparse(sizestr, NULL);
-  if (new_size == 0) {
+  new_size = memparse(sizestr, &retptr);
+  if (*retptr != '\0' || new_size == 0) {
    ret = -EINVAL;
    goto out_free;
   }
-- 
1.7.1